pkg/controller/actions/deploy/action_deploy_cache.go

```diff
@@ -10,27 +10,44 @@ import (
 // This code is heavily inspired by https://github.com/kubernetes-sigs/cluster-api/tree/main/internal/util/ssa
 
 const (
-	// ttl is the duration for which we keep the keys in the cache.
-	ttl = 10 * time.Minute
+	DefaultCacheTTL = 10 * time.Minute
 )
 
 type Cache struct {
-	s cache.Store
+	s   cache.Store
+	ttl time.Duration
 }
 
-func newCache() *Cache {
-	r := &Cache{
-		s: cache.NewTTLStore(func(obj interface{}) (string, error) {
+type CacheOpt func(*Cache)
+
+func WithTTL(ttl time.Duration) CacheOpt {
+	return func(c *Cache) {
+		c.ttl = ttl
+	}
+}
+
+func newCache(opts ...CacheOpt) *Cache {
+	c := Cache{
+		ttl: DefaultCacheTTL,
+	}
+
+	for i := range opts {
+		opts[i](&c)
```

Comment:
Wondering, why you are iterating over indexes vs values.  To avoid aliasing just in case?

Reply:
done

Reply:
changed it anyway :)